src/plugins/intel_gpu/src/graph/reorder.cpp

```diff
@@ -170,8 +170,11 @@ std::vector<layout> reorder_inst::calc_output_layouts(reorder_node const& /*node
 
     auto ifmt = input_layout.format;
     auto ofmt = desc->output_format == format::any ? ifmt : desc->output_format;
-
-    return { layout(input_layout.get<ShapeType>(), desc->output_data_types[0].value(), ofmt, desc->output_paddings[0]) };
+    auto ofdt = desc->output_data_types[0].value();
+    if (input_layout.is_dynamic()) {
+        return {layout(ShapeType::dynamic(format::dimension(desc->output_format)), ofdt, ofmt, desc->output_paddings[0])};
+    }
+    return { layout(input_layout.transform(ofmt), ofdt, ofmt, desc->output_paddings[0]) };
```

Comment:
Won't it break shape inference? out format rank can be greater than input dims rank, i.e. rank extension is done by this code. I think cases like `[2d shape] -> reorder (bfyx) -> gather` will fail as gather expected 2d shape on input, but received 4d

Reply:
My fix was to resolve some error in such a case :
- (5d) => reorder => (4d) where the reorder result was [1, 2, 3, 4, 5] with bfyx format. In such a case, I think the shape should be transformed for correct shape infer afterwords. 

But as you mentioned dim < 4d seems problematic. 
As is, we cannot ditinguish the two cases, where one actually want to reorder to 4d, or the 4d is just becuase it is minimal. 
We'll need to distinguish them. 
How about add reorder primitive to have additional information, other than the format...  like output_dim ...?  :(

Reply:
Why do we need to reorder from 5d format with 5d dims tensor to bfyx?

Reply:
1st, it happend in reorder_input. 
Actually for this matter, I think the reorder should not happen by checking only the output layout. 
(It is another problem) 

However, reorder happend with input (5d, bfzyx, {1, 2, 3, 4, 5})  and  output (4d, bfyx, {1, 2, 3, 4, 5} ) happened. Thgouth I think in this case the reorder itself should not happen, but still the reorder shape itself looks strange. Reorderd shape to 4d should be {1, 2, 12, 5} isnt it?